core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -703,9 +689,10 @@ public void endVerification(String seed) {
     @Override
     public RepositoryData getRepositoryData() {
         try {
-            RepositoryData repositoryData = updateIndexGenIfNecessary();
-            if (repositoryData == null) {
-                repositoryData = readIndexGen();
+            RepositoryData repositoryData = readIndexGen();
```

Comment:
I think if you collapse `readIndexGen` into `getRepositoryData` you will be able to remove otherwise unnecessary isLegacyFormat flag from RepositoryData.